src/cryptography/x509/extensions.py

```diff
@@ -48,18 +47,9 @@ def _key_identifier_from_public_key(public_key):
             serialization.Encoding.DER,
             serialization.PublicFormat.SubjectPublicKeyInfo
         )
-        spki, remaining = decoder.decode(
-            serialized, asn1Spec=_SubjectPublicKeyInfo()
-        )
-        assert not remaining
-        # the univ.BitString object is a tuple of bits. We need bytes and
-        # pyasn1 really doesn't want to give them to us. To get it we'll
-        # build an integer and convert that to bytes.
-        bits = 0
-        for bit in spki.getComponentByName("subjectPublicKey"):
-            bits = bits << 1 | bit
-
-        data = utils.int_to_bytes(bits)
+
+        spki = _SubjectPublicKeyInfo.load(serialized)
+        data = spki["subjectPublicKey"].cast(OctetBitString).native
```

Comment:
here we need just bytes though. should that be

`data = six.binary_type(PublicKeyInfo.load(serialized)['public_key'])` ?